### PR TITLE
[Internal] Usage: Add README.md for usage samples

### PR DESCRIPTION
**Pull request by tangerinous**
# [Internal] Usage: Add README.md for ApplicationInsights usage sample

## Description

Add README.md for ApplicationInsights usage sample

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+```PowerShell
+dotnet run
+```
+
+## Getting Started
+
+### Configure Application Insights Telemetry
+
+```C#
+// <SetUpApplicationInsights>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md:25]: These are related code references right? 
Actual run don't need to really change.

Comment (tangerinous) [reply Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md:25]: Should I remove code samples from the README and only leave descriptions and screenshots if necessar?

File: Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Cosmos client with Azure Function
+
+THis sample demonstrates how to maintain a Cosmos client and reuse the instance among Azure Function executions.
+
+## Prerequisites
+
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase and Container
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md:8]:
```suggestion
- Create a Database and Container
```

File: Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md:12]:
```suggestion
Before running the application you need fill out `EndpointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md:25]: Or cover it in a separate section/document.

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/README.md:25]: `</SetUpApplicationInsights>` is for MSDocs, please do not include any `<XXXXXXXXX>` comments in README snippets

File: Microsoft.Azure.Cosmos.Samples/Usage/CustomSerialization/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Cosmos Bulk support
+
+This sample demonstrates an example of configuring custom serialization
+
+## Description
+
+This example covers severale options:
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/CustomSerialization/README.md:7]:
```suggestion
This example covers several options:
```

File: Microsoft.Azure.Cosmos.Samples/Usage/CustomSerialization/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+- Async main requires c# 7.1 which is set in the csproj with the LangVersion attribute
+
+## Getting Started
+
+### Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/CustomSerialization/README.md:24]:
```suggestion
Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/CustomDiagnosticAndEventListener/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+- Async main requires c# 7.1 which is set in the csproj with the LangVersion attribute
+
+## Getting Started
+
+### Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/CustomDiagnosticAndEventListener/README.md:23]:
```suggestion
Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+- Async main requires c# 7.1 which is set in the csproj with the LangVersion attribute
+
+## Getting Started
+
+### Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/README.md:23]:
```suggestion
Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+## Prerequisites
+
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase and Container
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+- Async main requires c# 7.1 which is set in the csproj with the LangVersion attribute
+
+## Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/README.md:14]:
```suggestion
Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkExecutorMigration/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+## Prerequisites
+
+- Azure Cosmos DB NoSQL Account
+  - Create a DataBase and Container
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+
+## Run
+
+Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkExecutorMigration/README.md:13]:
```suggestion
Before running the application you need fill out `EndPointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+## Prerequisites
+
+- Azure Cosmos DB NoSQL Account
+- Create a Database and Container
+- Microsoft.Azure.Cosmos NuGet [package](http://www.nuget.org/packages/Microsoft.Azure.Cosmos/)
+
+## Run
+
+Before running the application you need fill out `EndpointUrl` and `AuthorizationKey` params in the [AppSettings.json](AppSettings.json)
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md:13]:
```suggestion
Before running the application you need fill out `EndpointUrl` and `AuthorizationKey` params in the [AppSettings.json](../appSettings.json)
```

File: Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# Cosmos client with Azure Function
+
+This sample demonstrates how to maintain a Cosmos client and reuse the instance among Azure Function executions.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/AzureFunctions/README.md:3]:
```suggestion
This sample demonstrates how to maintain a Cosmos client instance following Singleton pattern and reuse the instance across Azure Function executions.
```

**Review by ealsur — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Comment by kirankumarkolli** [comment]
@tangerinous can you please accept the cla license for PR to merge?

**Comment by tangerinous** [comment]
@microsoft-github-policy-service agree

**Pull request merged**